t/input.t

```diff
@@ -1289,3 +1289,23 @@ X-Forwarded-For: 8.8.8.8
 Foo: 127.0.0.1
 --- no_error_log
 [error]
+
+
+
+=== TEST 50: clear input headers with wildcard
+--- config
+    location /hello {
+        more_clear_input_headers 'X-Hidden-*';
+        content_by_lua '
+            ngx.say("X-Hidden-One: ", ngx.var.http_x_hidden_one)
+            ngx.say("X-Hidden-Two: ", ngx.var.http_x_hidden_two)
+        ';
+    }
+--- request
+    GET /hello
+--- more_headers
+X-Hidden-One: i am hidden
+X-Hidden-Two: me 2
+--- response_body
+X-Hidden-One: nil
+X-Hidden-Two: nil
```

Comment:
It "worked" in the sense that if two headers like `X-Hidden-*` were set, and the wildcard was used with `more_set_input_headers` then _only the first header_ would be set to that value.

I think that is just weird, and completely not intentional, so I changed the code to prevent this (and added a test to make sure).